proposals/20230323-end-to-end-testing.md

```diff
@@ -0,0 +1,372 @@
+# End-to-end Testing
+
+**Authors**: Dominik Fleischmann ([@domFleischmann](https://github.com/domFleischmann)),
+Kimonas Sotirchos ([@kimwnasptd](https://github.com/kimwnasptd)),
+and Anna Jung ([@annajung](https://github.com/annajung))
+
+## Background
+
+Previously, the Kubeflow community leveraged prow optional-test-infra for e2e testing with credit from AWS. After the
+optional test infrastructure deprecation notice, all WGs moved their test to GitHub Actions as a temporary solution. Due
+to resource constraints of GitHub-hosted runners, the Kubeflow community stopped supporting e2e tests as part of the
+migration. In partnership with Amazon, a new AWS account has been created with sponsored credits. With the new AWS
+account, the Kubeflow community is no longer limited by resource constraints posed by GitHub Actions. To enable the e2e
+test for the Manifest repo, this doc proposes a design to set up the infrastructure needed to run the necessary tests.
+
+References
+
+- [Optional Test Infra Deprecation Notice](https://github.com/kubeflow/testing/issues/993)
+- [Alternative solution to removal of test on optional-test-infra](https://github.com/kubeflow/testing/issues/1006)
+
+## Goal
+
+Enable the e2e testing for the Manifest repo and leverage it to shorten the manifest testing phase of the Kubeflow
+release cycle and to increase quality of the Kubeflow release by ensuring Kubeflow components and dependencies work
+correctly together.
+
+## Proposal
+
+After some initial conversations, it has been agreed to create integration tests based on GitHub Actions, which will
+spawn an EC2 instance with enough resources to deploy the complete Kubeflow solution and run some end-to-end testing.
+
+### Alternative solutions considered
```

Comment:
Can we move this section to the end of the doc? IMO it's more clear if the proposal is focused only on what we propose, and then at the end expose some info on the alternatives. So the reader's focus is always on the selected approach while going through the doc

Reply:
Moved to the last section